### ISSUE TITLE
make embark-live after exiting the minibuffer

### ISSUE THREAD

**Issue by shelper** [oantolin/embark#503]
I am using vertico for minibuffer autocompletion. 
after i got a list of results in minibuffer, 
i invoke embark and get the embark-live buffer, but the minibuffer still keeps open, if I press ctrl-g to exit minibuffer, then the embark-live buffer also got hidden...

I think it would be much smoother experience if the embark-live buffer is kept in focus when invoked, and minibuffer closes automatically.
At least for other things like embark-collection or embark-export, when invoked, what happened is the minibuffer will close and i have the exported buffer ready for me to explore. Or did i miss anything?

**Comment by oantolin** [comment]
embark-live keeps the minibuffer open so that you can keep updating the minibuffer contents and see how the results change. After you quit the minibuffer Emacs's usual window configuration restoration kicks and this hides the Embark live buffer, but it's still alive.

If you already use Vertico, may I ask why you wanted to use embark-live from the minibuffer rather than embark-collect?

**Comment by shelper** [comment]
thanks for the explanation, 
the embark-live  provides a way to follow and preview the results in the source files

the embark-collect does not provide this following feature.

**Comment by shelper** [comment]
I just tested and found the embark-live does not get updated when i change the search string.....

**Comment by oantolin** [comment]
Oh, that's interesting! It works fine here, so I'd love some help reproducing the problem.

**Comment by oantolin** [comment]
Just to be clear: if you call `embark-live` from the minibuffer it is supposed to auto-update only while the minibuffer is open, and to stop updating once you quit the minibuffer.

Also, while using it from the minibuffer does work it's not really the intended use case. The idea is that you use it from a normal buffer to get a live updating table of contents. If you have consult and embark-consult installed, try for example using `embark-live` from an Emacs Lisp buffer or an Org mode buffer.

**Comment by oantolin** [comment]
> now I understand why the minibuffer does not go away when embark-live invoked, but is there a way to avoid hiding the embark live buffer?

To make sure I understand, you are asking about it being hidden once you kill the minibuffer? If you are on Emacs 28 or higher you can set `read-minibuffer-restore-windows` to `nil`.

You can also simply switch to the embark live buffer after quitting the minibuffer.

**Comment by oantolin** [comment]
> the embark-live provides a way to follow and preview the results in the source files

I'm not sure what you mean here. What "preview" are you talking about?

If you mean the previews implemented by the consult package, then you don't need `embark-live` for them as you also get them with Vertico.

If you mean instead that you are using actions on the candidates in the embark-live buffer to "preview" things, then that also works perfectly fine with embark-collect.

**Comment by oantolin** [comment]
I want to help, but in general I'm just pretty confused about what's going on and what you want to happen instead. :) But if you explain it I'm sure we can figure things out together.

**Comment by minad** [comment]
See also https://github.com/minad/vertico/wiki#automatically-shrink-vertico-for-embark-live

**Comment by minad** [comment]
If it is only about live preview in a buffer you can also try vertico-buffer. It is also unclear to me what @shelper tries to achieve.

**Comment by shelper** [comment]
thanks for the notes, and sorry for the confusion
I am using Doom emacs with vertico enabled. not much personal settings, if you suggest this is an issue with doom, i will file an issue there.

basically What i want to do is
1. search my project using ripgrep
2. get a list of result and export to embark-live buffer
3. quickly preview the results in their  original locations/files

setting `read-minibuffer-restore-windows` to `nil` does help, now i can see the embark-live buffer after closing the minibuffer.
but i still dont get live update in embark-live buffer when i input into the minibuffer for new search ,
here is what i did
1. search project using ripgrep for `foo`
2. invoke embark-live
3. change search string from `foo` to `bar`
4. i dont see update in my embark-live window

see screen shot below:
![image](https://user-images.githubusercontent.com/546163/166935006-b982ea75-181b-4156-9414-68c332e78147.png)

**Comment by oantolin** [comment]
OK, so first, the `embark-live` buffer contains the same results that Vertico is show you so seeing the `embark-live` buffer is probably not useful is it? 

Second, what are you doing for preview? Switching over to the `embark-live` buffer and moving around there, so that as you move around automatically each file is opened at the correct location, in another window? If so, that is `consult-preview-at-point-mode` and (1) it will stop working after you've closed the minibuffer, but (2) more importantly: you don't need `embark-live` for this at all!, you can just scroll through the search results in Vertico to get the previews! (So you see why I'm confused about why you think you need to use `embark-live`.)

Finally, I do see the issue you mention of results not updating, I didn't realize there was this problem before. It happens with consult's async search commands, when `embark-live` sees you typed and updates but does it too soon, *before* the async search has returned its new results! One way you can get the display to update is to make a change to the input that does not change the results, in your example of changing `#foo` to `#bar`, you could add a second `#` to get `#bar#`. The results should update then.

I'm sorry for insisting on this point but why do you want to use `embark-live` to preview the results in their original locations. Is preview within Vertico's list of results not working for you?

**Comment by shelper** [comment]
you are right, just need sometime to get familiar with vertico and embark... green to emacs and it is so different (and fun) compared other ide & editors...

thanks for the clarification, i will close this issue!

**Comment by oantolin** [comment]
So does preview with just Vertico work for you then? If not, I may be able to help, and in the worst case scenario we can summon the expert.

**Comment by shelper** [comment]
i did some hack to the preview so i use c-j/k to move around the results in minibuffer and c-n/p to move around the results + preview at point, 
I am good with the setup now,  your explanation definitely helps!
